Header.c

```diff
@@ -150,35 +150,71 @@ void Header_reinit(Header* this) {
 }
 
 void Header_draw(const Header* this) {
-   int height = this->height;
-   int pad = this->pad;
+   const int height = this->height;
+   const int pad = this->pad;
    attrset(CRT_colors[RESET_COLOR]);
    for (int y = 0; y < height; y++) {
       mvhline(y, 0, ' ', COLS);
    }
-   int width = COLS / this->nrColumns - (pad * this->nrColumns - 1) - 1;
+   const int width = COLS / this->nrColumns - (pad * this->nrColumns - 1) - 1;
    int x = pad;
 
    Header_forEachColumn(this, col) {
       Vector* meters = this->columns[col];
       for (int y = (pad / 2), i = 0; i < Vector_size(meters); i++) {
          Meter* meter = (Meter*) Vector_get(meters, i);
-         meter->draw(meter, x, y, width);
+
+         int actualWidth;
+         if (meter->mode == TEXT_METERMODE)
+            actualWidth = meter->columnWidthCount * width + (meter->columnWidthCount - 1) * (2 * pad + 1);
+         else
+            actualWidth = width;
+
+         meter->draw(meter, x, y, actualWidth);
          y += meter->h;
       }
       x += width + pad;
    }
 }
 
+/*
+ * Calculate how many columns the current meter is allowed to span,
+ * by counting how many columns to the right are empty or contain a BlankMeter.
+ * Returns the number of columns to span, i.e. if the direct neighbor is occupied 1.
+ */
+static int calcColumnWidthCount(const Header* this, const Meter* curMeter, const int pad, const int curColumn, const int curHeight) {
+   for (int i = curColumn + 1; i < this->nrColumns; i++) {
+      const Vector* meters = this->columns[i];
+
+      int height = pad;
+      for (int j = 0; j < Vector_size(meters); j++) {
+         const Meter* meter = (const Meter *) Vector_get(meters, j);
+
+         if (height >= curHeight + curMeter->h)
+            break;
+
+         height += meter->h;
+         if (height <= curHeight)
+            continue;
```

Comment:
I'm not sure this successfully catches, if two meters are offset by a few rows.

```
+---+---+
|   | 3 |
+ 1 +---+
|   |   |
+---+ 4 +
| 2 |   |
+---+---+
```

Question related to when checking 1 if non-blank 4 is caught?
Also what of 2 if 4 were even one line higher?

Reply:
I tested several cases and all seems to work.